pkg/cmd/version/version.go

```diff
@@ -32,16 +33,35 @@ the version was built from, the build time, and the version of Go it was
 compiled with.`
 )
 
+const versionNotifFlag = "show-update-notifications"
+
+type versionOpts struct {
+	shouldShowNotifications bool
+}
+
 func VersionCommand() *cobra.Command {
+	opts := &versionOpts{}
 
 	cmd := &cobra.Command{
 		Use:   "version",
 		Short: shortDesc,
 		Long:  longDesc,
 		Run: func(cmd *cobra.Command, args []string) {
-			output.Infof("Version: %s\nCommit: %s\nBuilt: %s\nGo version: %s\n", constants.Version, constants.GitCommit, constants.BuildTime, constants.GoVersion)
+			if cmd.Flags().Changed(versionNotifFlag) {
+				configHome, ok := cmd.Context().Value(constants.ConfigKey{}).(string)
+				if !ok {
+					output.Fatalln("default config path not set on command context")
+				}
+				if err := update.SetShowNotifications(configHome, opts.shouldShowNotifications); err != nil {
+					output.Fatalln(err)
+				}
```

Comment:
You can possibly move the code for handling the flag to its own function.

Reply:
I could, but it would only reduce this section from 7 lines to 3 (call complete and check the error). This way it's simple to suppress the default output for `kit version` when the option is set.